File: VB/WcfService1/Service1.svc.vb
﻿Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.Xpo
Imports DevExpress.Xpo.DB

Namespace WcfService1
	Public Class Service1
		Inherits DataStoreService
		Public Shared DataStore As IDataStore
		Shared Sub New()
			Dim connectionString As String = MSSqlConnectionProvider.GetConnectionString("localhost", "ServiceDB")
			DataStore = XpoDefault.GetConnectionProvider(connectionString, AutoCreateOption.DatabaseAndSchema)
		End Sub
		Public Sub New()
			MyBase.New(DataStore)
		End Sub
	End Class
End Namespace

File: CS/WcfService1/Service1.svc.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.Xpo;
using DevExpress.Xpo.DB;

namespace WcfService1 {
    public class Service1 : DataStoreService {
        public static IDataStore DataStore;
        static Service1() {
            string connectionString = MSSqlConnectionProvider.GetConnectionString("localhost", "ServiceDB");
            DataStore = XpoDefault.GetConnectionProvider(connectionString, AutoCreateOption.DatabaseAndSchema);
        }
        public Service1()
            : base(DataStore) {
        }
    }
}
